### DISCUSSION TITLE
How to add force at one point?

### DISCUSSION THREAD

**Discussion by Aristomo** [google-deepmind/mujoco#641] [Asking for Help]
I have a robot and I want to add forces at one point on this robot. Just like there is a hand to push it, but I do not want others see it. Ho w to do this? I wish the point where will be push can be given according to the coordinate. Thanks!

**Comment by yuvaltassa** [comment]
Write into xfrc_applied to apply the force/torque at the body com, or use mj_applyFT is you want an arbitrary point.

**Reply by Kcygt** [reply]
Hi,
Do you have any example for mj_applyFT ?
Thank you

**Reply by yuvaltassa** [reply]
Isn't there something in the tutorial? Maybe not

**Reply by Kcygt** [reply]
I don't know which tutorial you referenced.  There are a number of videos on this link ( **_https://www.youtube.com/watch?v=j1nCeqtfySQ_** ), but I didn't notice any examples. Could you share any other tutorials you may know of?

**Reply by yuvaltassa** [reply]
I meant [the introductory tutorial](https://github.com/google-deepmind/mujoco?tab=readme-ov-file#getting-started), of course. Not clear how you missed it, that's a pretty prominent location. Do you think there is a place it should be mentioned where it isn't? It's also [here](https://mujoco.readthedocs.io/en/latest/python.html#tutorial-notebook), so I don't know where else I can put it...

That said, mj_applyFT is not mentioned in the tutorial. In the second cell under "Simulation", you would do something like:
```py
frames = []
mujoco.mj_resetData(model, data)
bodyid = model.body('box_and_sphere')
while data.time < duration:
  mujoco.mj_applyFT(model, data, my_force, my_torque, point_on_body, bodyid, data.qfrc_applied)
  mujoco.mj_step(model, data)
  if len(frames) < data.time * framerate:
    renderer.update_scene(data, scene_option=scene_option)
    pixels = renderer.render()
    frames.append(pixels)
```

**Reply by Kcygt** [reply]
That's useful. Thank you.
